File: java/src/main/java/com/epam/eng/pattern/builder/PizzaOrder.java
package com.epam.eng.pattern.builder;

public class PizzaOrder {
    private String pizzaType;
    private String size;
    private String crustType;
    private boolean extraCheese;
    private boolean pepperoni;
    private boolean mushrooms;

    public void setPizzaType(String pizzaType) {
        this.pizzaType = pizzaType;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public void setCrustType(String crustType) {
        this.crustType = crustType;
    }

    public void setExtraCheese(boolean extraCheese) {
        this.extraCheese = extraCheese;
    }

    public void setPepperoni(boolean pepperoni) {
        this.pepperoni = pepperoni;
    }

    public void setMushrooms(boolean mushrooms) {
        this.mushrooms = mushrooms;
    }
}

File: ts/builder/PizzaOrder.ts
class PizzaOrder {
    private pizzaType: string = "";
    private size: string = "";
    private crustType: string = "";
    private extraCheese: boolean = false;
    private pepperoni: boolean = false;
    private mushrooms: boolean = false;

    public setPizzaType(pizzaType: string) {
        this.pizzaType = pizzaType;
    }

    public setSize(size: string) {
        this.size = size;
    }

    public setCrustType(crustType: string) {
        this.crustType = crustType;
    }

    public setExtraCheese(extraCheese: boolean) {
        this.extraCheese = extraCheese;
    }

    public setPepperoni(pepperoni: boolean) {
        this.pepperoni = pepperoni;
    }

    public setMushrooms(mushrooms: boolean) {
        this.mushrooms = mushrooms;
    }
}

var order = new PizzaOrder();
console.log(order);

// todo: consider create a PizzaOrderBuilder so that can build PizzaOrder step by step.
